WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Plugins/SitePluginListViewModelTests.swift

```diff
@@ -174,11 +174,183 @@ class PluginListViewModelTests: XCTestCase {
+        let storesManager = MockPluginStoresManager(shouldSucceed: false, completionDelay: delay)
+        let viewModel = PluginListViewModel(siteID: sampleSiteID, storesManager: storesManager)
+
+        // When
+        viewModel.resyncPlugins {}
+
+        // Then
+        XCTAssertEqual(viewModel.pluginListState, .syncing)
+        DispatchQueue.main.asyncAfter(deadline: .now() + delay) {
+            XCTAssertEqual(viewModel.pluginListState, .error)
+            expect.fulfill()
+        }
+        waitForExpectations(timeout: Constants.expectationTimeout, handler: nil)
+    }
 }
 
 private extension PluginListViewModelTests {
     func insert(_ readOnlyPlugin: Yosemite.SitePlugin) {
         let plugin = storage.insertNewObject(ofType: StorageSitePlugin.self)
         plugin.update(with: readOnlyPlugin)
+        storage.saveIfNeeded()
+    }
+
+    func updateStorage(with readOnlyPlugin: Yosemite.SitePlugin) {
+        guard let plugin = storage.loadPlugin(siteID: readOnlyPlugin.siteID, name: readOnlyPlugin.name) else {
+            return
+        }
+        plugin.update(with: readOnlyPlugin)
+        storage.saveIfNeeded()
+    }
+}
+
+final class MockPluginStoresManager: DefaultStoresManager {
+    /// Whether synchronizePlugins action was triggered.
+    ///
+    var invokedSynchronizePlugins = false
+
+    /// The site ID that the `synchronizePlugins` action was triggered with.
+    ///
+    var invokedSynchronizePluginsWithSiteID: Int64 = 0
+
+    /// Whether the mock `synchronizePlugins` action handler should succeed.
+    ///
+    private let shouldSucceed: Bool
+
+    /// Delay time for completion block of the mock `synchronizePlugins` action.
+    ///
+    private let completionDelay: TimeInterval
+
+    enum MockPluginError: Error {
+        case mockError
+    }
+
+    init(shouldSucceed: Bool, completionDelay: TimeInterval = 0) {
+        self.shouldSucceed = shouldSucceed
+        self.completionDelay = completionDelay
+        let sessionManager = SessionManager.testingInstance
+        sessionManager.setStoreId(134)
+        super.init(sessionManager: sessionManager)
+    }
+
+    // MARK: - Overridden Methods
+    override func dispatch(_ action: Action) {
+        if let sitePluginAction = action as? SitePluginAction {
+            onPluginAction(sitePluginAction)
+        }
+    }
+
+    private func onPluginAction(_ action: SitePluginAction) {
+        switch action {
+        case .synchronizeSitePlugins(let siteID, let onCompletion):
+            invokedSynchronizePlugins = true
+            invokedSynchronizePluginsWithSiteID = siteID
+            DispatchQueue.main.asyncAfter(deadline: .now() + completionDelay) { [weak self] in
+                if self?.shouldSucceed == true {
+                    onCompletion(.success(()))
+                } else {
+                    onCompletion(.failure(MockPluginError.mockError))
+                }
+            }
+        }
```

Comment:
I think we have a `MockStoresManager` that can help you achieve the same as this class.

EG:
https://github.com/woocommerce/woocommerce-ios/blob/4fe952211777fad134084cbc19d4db229a37c8c9/WooCommerce/WooCommerceTests/ViewRelated/Orders/Issue%20Refund/RefundConfirmationViewModelTests.swift#L109-L122

Reply:
Great idea! I have migrated to this and it looks so much neater 🎉